atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/bucketingthings/DefaultSweepAssignedBucketStoreTest.java

```diff
@@ -162,4 +170,106 @@ public void updateStartingBucketForShardFailsAfterTooManyAttempts() {
         assertThatThrownBy(() -> store.updateStartingBucketForShardAndStrategy(bucket))
                 .isInstanceOf(CheckAndSetException.class);
     }
+
+    @Test
+    public void getSweepableBucketsReturnsTimestampRangesforPresentBucketsForEachShardUpToTwoHundredBuckets() {
+        ShardAndStrategy shardAndStrategy = ShardAndStrategy.of(12, SweeperStrategy.THOROUGH);
+        List<SweepableBucket> buckets = IntStream.range(0, 400)
+                .mapToObj(i -> SweepableBucket.of(Bucket.of(shardAndStrategy, i), TimestampRange.of(i, i + 1)))
+                .collect(Collectors.toList());
+        buckets.forEach(
+                bucket -> store.putTimestampRangeForBucket(bucket.bucket(), Optional.empty(), bucket.timestampRange()));
+
+        assertThat(store.getSweepableBuckets(Set.of(Bucket.of(shardAndStrategy, 0))))
+                .containsExactlyInAnyOrderElementsOf(buckets.subList(0, 200));
+    }
+
+    @Test
+    public void getSweepableBucketsSkipsBucketsWithMajorIdentifierBeforeStartBucketProvided() {
+        ShardAndStrategy shardAndStrategy = ShardAndStrategy.of(12, SweeperStrategy.THOROUGH);
+        List<SweepableBucket> buckets = IntStream.range(0, 200)
+                .mapToObj(i -> SweepableBucket.of(Bucket.of(shardAndStrategy, i), TimestampRange.of(i, i + 1)))
+                .collect(Collectors.toList());
+
+        buckets.forEach(
+                bucket -> store.putTimestampRangeForBucket(bucket.bucket(), Optional.empty(), bucket.timestampRange()));
+
+        assertThat(store.getSweepableBuckets(Set.of(Bucket.of(shardAndStrategy, 100))))
+                .containsExactlyInAnyOrderElementsOf(buckets.subList(100, 200));
+    }
+
+    @Test
+    public void getSweepableBucketsReturnsLessThanMaxBucketLimitIfLessThanTwoHundredBucketsPresent() {
+        ShardAndStrategy shardAndStrategy = ShardAndStrategy.of(12, SweeperStrategy.THOROUGH);
+        List<SweepableBucket> buckets = IntStream.range(0, 10)
```

Comment:
optional: should we parameterize on the number and get `0`, `1`, `12` and `199` buckets? Might be overkill

Reply:
Ha, sure, will do!